centreon-dsm/www/modules/centreon-dsm/core/configuration/services/DB-Func.php

```diff
@@ -83,30 +93,38 @@ function getListServiceForPool($poolId)
  *
  * @param int $hostId The host id
  * @param string $poolPrefix The pool prefix
+ * @param int|null $poolId The pool id (optional)
  * @return bool
  */
 function hostPoolPrefixUsed($hostId, $poolPrefix, $poolId = null)
 {
     global $pearDB;
 
-    $query = "SELECT COUNT(pool_id) AS nb FROM mod_dsm_pool WHERE pool_host_id = '" .
-        $hostId . "' AND pool_prefix = '" . $poolPrefix . "'";
+    $query = "SELECT COUNT(pool_id) AS nb FROM mod_dsm_pool WHERE pool_host_id = :host_id AND pool_prefix = :pool_prefix";
+    $params = [
+        ':host_id' => [$hostId, PDO::PARAM_INT],
+        ':pool_prefix' => [$poolPrefix, PDO::PARAM_STR]
+    ];
+
     if (!is_null($poolId)) {
```

Comment:
you can cast $poolId into int before the test, and your test can be : `if($poolId <= 0)` it's safer